android/app/src/main/java/edu/berkeley/eecs/e_mission/data_sync/ConfirmTripsAdapter.java

```diff
@@ -62,118 +105,302 @@ public ConfirmTripsAdapter(Context context, boolean autoInitialize) {
+
+			if (!OnboardingActivity.getOnboardingComplete(cachedContext)) {
+				generateNotification(CONFIRM_TRIPS_ID, "Finish setting up app", edu.berkeley.eecs.e_mission.OnboardingActivity.class);
+				return;
+			}
+
+			System.out.println("Can we use the extras bundle to transfer information? " + extras);
+			// Get the list of uncategorized trips from the server
+			// hardcoding the URL and the userID for now since we are still using fake data
+			String userName = UserProfile.getInstance(cachedContext).getUserEmail();
+			System.out.println("real user name = " + userName);
+
+			if (userName == null || userName.trim().length() == 0) {
+				System.out.println("we don't know who we are, so we can't get our data");
+				return;
+			}
+			// First, get a token so that we can make the authorized calls to the server
+			String userToken = GoogleAccountManagerAuth.getServerToken(cachedContext, userName);
+
+			// TODO: Use ORM to convert this to a Section[]
+			try {
+				// First push and then pull, so that we don't end up re-reading data that the user has just classified
+				JSONArray classifiedSections = convertListToJSON(dbHelper.getAndDeleteClassifiedSections());
+				statsHelper.storeMeasurement(cachedContext.getString(R.string.sync_push_list_size),
+						String.valueOf(classifiedSections.length()), syncTs);
+				// System.out.println("classifiedSections = "+classifiedSections);
+				if (classifiedSections.length() > 0) {
+					CommunicationHelper.pushClassifications(cachedContext, userToken, classifiedSections);
+				} else {
+					System.out.println("No user classified sections, skipping push to server");
+				}
+
+				/*
+				 * Originally, we just added the classified sections to the database. But this gives rise
+				 * to synchronization errors if the user is accessing the data from multiple devices.
+				 * This is probably not a huge issue initially for most people, but some cases that
+				 * I can think of are:
+				 * - two people sharing the same account like Tom and me
+				 * - people using a different device for the data collection and a different one
+				 * (google watch) for the classification.
+				 *
+				 * The database on the server is the source of truth, so we can always just synchronize to it.
+				 * So let's clear our local cached copy before pulling data from the server.
+				 */
+
+				if (Service.equals("Ours")) {
+					dbHelper.clear();
+				}
+
+				JSONArray unclassifiedSections = CommunicationHelper.getUnclassifiedSections(
+						cachedContext, userToken);
+				statsHelper.storeMeasurement(cachedContext.getString(R.string.sync_pull_list_size),
+						String.valueOf(unclassifiedSections.length()), syncTs);
+				// save the unclassified sections to the database
+				if (Service.equals("Ours")) {
+					dbHelper.storeNewUnclassifiedTrips(convertJSONToList(unclassifiedSections));
+				} else if (Service.equals("CouchBase")) {
+					Document document = database.createDocument();
```

Comment:
I am confused here. Our code does not only include the save to the database `dbHelper.....`, but also the REST API call to read the section `CommunicationHelper.getUnclassifiedSections....". I thought that in the couchbase code, the sync would happen automatically, so that we would not need to make the REST API call.

So given that you make the REST API call whether `Service` is `Ours` or `Couchbase`, I am not sure whether the Couchbase update is actually working or not.